src/router.ts

```diff
@@ -98,19 +98,30 @@ router.post("/:target", async (req: Request, res: Response) => {
             let sent = await EmailService.sendMail(req.params.target, from, fieldSubject, fieldBody, files);
 
             if(sent instanceof Error || !sent) {
-                if(target.redirect?.error) return res.redirect(target.redirect.error);
+                if(target.redirect?.error) return res.redirect(getRedirectUrl(req, target.redirect.error));
                 return res.status(500).send({ message: (<Error>sent).message }).end();
             }
 
             if(target.redirect?.success) {
-                return res.redirect(target.redirect.success);
+                return res.redirect(getRedirectUrl(req, target.redirect.success));
             }
 
             return res.status(200).end();
         }
     });
 });
 
+function getRedirectUrl(req: Request, targetRedirectUrl: string) {
+    let redirectUrl = targetRedirectUrl
+
+    const urlPattern = /^(https?):\/\//;
+    if(!urlPattern.test(redirectUrl)) {
+        redirectUrl = (req.header('Referer') || '/') + redirectUrl;
+    }
+
+    return redirectUrl
+}
+
```

Comment:
Please move that to a separate file. The router should only contain express routes, not utility functions.

Reply:
OK, i will fix it tomorrow. do you prefer 'util/redirect.ts' or Just 'util.ts' in Root?

Reply:
> OK, i will fix it tomorrow. do you prefer 'util/redirect.ts' or Just 'util.ts' in Root?

'utils/redirect.ts' is fine. Thank you :)

Reply:
getRedirectUrl is now extracted to util/redirect.ts